runtime/major_gc.c

```diff
@@ -44,8 +44,8 @@
 #define INITIAL_POOLS_TO_RESCAN_LEN 4
 
 typedef struct {
-  value block;
-  uintnat offset;
+  value* start;
+  value* end;
```

Comment:
IIUC `end` points to the word after the last markable word? If so, perhaps worth adding a comment here.